atomic_physics/tests/test_utils.py

```diff
@@ -1,275 +1,205 @@
         )
-        shift = ac_zeeman(Omega, w_spectator, w_rf)
 
-        np.testing.assert_allclose(
-            shift,
-            ac_zeeman_shift_for_state(
-                atom=ion,
-                state=victim_state_index,
-                drive=RFDrive(
-                    frequency=w_rf,
-                    amplitude=rf_amplitude,
-                    polarization=polarization.SIGMA_MINUS_POLARIZATION,
-                ),
-            ),
+        rf_rsb_sigma = RFDrive(
+            frequency=f - f_mode,
+            amplitude=B_rf,
+            polarization=SIGMA_MINUS_POLARIZATION + SIGMA_PLUS_POLARIZATION,
         )
 
-        # sigma+ shift
-        shift = 0.0
-        for spectator_F, spectator_M in ((3, +1), (4, -1), (4, 1)):
-            spectator_state_index = ion.get_state_for_F(
-                level=ca43.S12, F=spectator_F, M_F=spectator_M
+        for qubit in qubit_pairs_and_shifts:
+            idx0 = ion.get_state_for_F(level, qubit[0][0], qubit[0][1])
+            idx1 = ion.get_state_for_F(level, qubit[1][0], qubit[1][1])
+
+            acz_diff_bsb_pi = ac_zeeman_shift_for_transition(
+                ion, [idx0, idx1], rf_bsb_pi
             )
-            Omega = ion.get_rabi_m1(
-                lower=spectator_state_index,
-                upper=victim_state_index,
-                amplitude=rf_amplitude,
+
+            acz_diff_rsb_pi = ac_zeeman_shift_for_transition(
+                ion, [idx0, idx1], rf_rsb_pi
             )
-            w_spectator = ion.get_transition_frequency_for_states(
-                states=(spectator_state_index, victim_state_index)
+
+            acz_diff_bsb_sigma = ac_zeeman_shift_for_transition(
+                ion, [idx0, idx1], rf_bsb_sigma
             )
-            shift += ac_zeeman(Omega, w_spectator, w_rf)
 
-        np.testing.assert_allclose(
-            shift,
-            ac_zeeman_shift_for_state(
-                atom=ion,
-                state=victim_state_index,
-                drive=RFDrive(
-                    frequency=w_rf,
-                    amplitude=rf_amplitude,
-                    polarization=polarization.SIGMA_PLUS_POLARIZATION,
-                ),
-            ),
-        )
+            acz_diff_rsb_sigma = ac_zeeman_shift_for_transition(
+                ion, [idx0, idx1], rf_rsb_sigma
+            )
 
-    def test_ac_zeeman_D_level(self):
-        """Check AC Zeeman shift calculations using a qubit in the metastable excited
-        D5/2 level in 137Ba+.
+            np.testing.assert_allclose(
+                [
+                    acz_diff_bsb_pi,
+                    acz_diff_rsb_pi,
+                    acz_diff_bsb_sigma,
+                    acz_diff_rsb_sigma,
+                ],
+                qubit[2],
+                rtol=1e-3,
+            )
 
-        This test is a bit more interesting than the ground-level tests because there
-        are more spectator transitions around
+    def test_acz_trap_rf_ca43(self):

```

Comment:
I'd probably be inclined to merge this with the previous test so it's just one long set of tests comparing Ca43+ values to my thesis.